core/src/networked_server.cpp

```diff
@@ -0,0 +1,176 @@
+#include <knoting/engine.h>
+#include <knoting/networked_server.h>
+
+namespace knot {
+NetworkedServer::~NetworkedServer() {}
+
+NetworkedServer::NetworkedServer(Engine& engine) : m_engine(engine), m_server(nullptr) {}
+void NetworkedServer::on_awake() {
+    uint8_t privateKey[KeyBytes] = {0};
+    m_server = std::make_shared<Server>(GetDefaultAllocator(), privateKey, Address(SERVER_ADDRESS, SERVER_PORT),
+                                        m_config, gameAdapter, get_time());
+    m_server->Start(MAX_CLIENTS);
+    log::debug("STARTED SERVER...");
+    seq.fill(0);
+    cliSeq.fill(0);
+    cliAck.fill(0);
+}
+void NetworkedServer::on_update(double m_delta_time) {
+    if (!m_server->IsRunning()) {
+        log::debug("server not running!");
+        return;
+    }
+
+    m_server->AdvanceTime(get_time());
+    m_server->ReceivePackets();
+
+    m_tickTime += m_delta_time;
+    if (m_server->GetNumConnectedClients() > 0) {
+        handle_recieved_packets(m_delta_time);
+        send_message();
+    } else {
+        seq.fill(0);
+    }
+    if (m_tickTime >= TICK) {
+        m_tickTime -= TICK;
+    }
```

Comment:
Do we need to decrease the tick time in here?

Reply:
where else would it go? in it's own function?